src/Workspaces/Core/Portable/Workspace/Workspace_Events.cs

```diff
@@ -12,12 +13,15 @@ namespace Microsoft.CodeAnalysis
     public abstract partial class Workspace
     {
         private readonly EventMap _eventMap = new EventMap();
+
         private const string WorkspaceChangeEventName = "WorkspaceChanged";
         private const string WorkspaceFailedEventName = "WorkspaceFailed";
         private const string DocumentOpenedEventName = "DocumentOpened";
         private const string DocumentClosedEventName = "DocumentClosed";
         private const string DocumentActiveContextChangedName = "DocumentActiveContextChanged";
 
+        private volatile bool _listenerInitialized = false;
```

Comment:
Why the volatile? You're not doing an interlocked tricks, and it's still under a lock, right?

Reply:
👍 .  Volatile is very hard to get right and feels like a mjor footgun.

Reply:
ya. it is under lock so don't need volatile.

Reply:
this no longer exist